src/WebJobs.Extensions/Extensions/Files/Config/FilesJobHostConfigurationExtensions.cs

```diff
@@ -57,16 +59,45 @@ public FilesExtensionConfig(FilesConfiguration filesConfig)
                 _filesConfig = filesConfig;
             }
 
+            private FileInfo GetFileInfo(FileAttribute attribute)
+            {
+                string boundFileName = attribute.Path;
+                string filePath = Path.Combine(_filesConfig.RootPath, boundFileName);
+                FileInfo fileInfo = new FileInfo(filePath);
+                return fileInfo;
+            }
+
+            public FileStream GetFileStream(FileAttribute attribute)
+            {
+                var fileInfo = GetFileInfo(attribute);
+                if ((attribute.Access == FileAccess.Read) && !fileInfo.Exists)
+                {
+                    return null;
```

Comment:
Letting the FileInfo.Open() call below call an exception below may provide a more clear error message than returning a null value for the Stream.

Reply:
Looking at your unit tests on the other PR, it appears we want to return null here since there isn't a reliable exception for the file not existing. Adding a comment here would help make that more clear.

Reply:
Right, null is the contract for "does not exist"